iree/turbine/kernel/wave/cache.py

```diff
@@ -206,6 +223,13 @@ def store_kernel_to_file(
         _write_file(cur_module_path, "w", module_str)
         kernel_sig_str = json.dumps([usage.name for usage in kernel_sig])
         _write_file(cur_kernelsig_path, "w", kernel_sig_str)
+        cur_hsaco_path = glob.glob(str(WAVE_RUNTIME_DIR / "*.hsaco"))
+        # Copy the hsaco file to the cache directory only if it exists.
+        if cur_hsaco_path:
+            cur_hsaco_path = cur_hsaco_path[0]
+            shutil.copy(cur_hsaco_path, cur_cache_basefile.with_suffix(".hsaco"))
+        cur_pkl_path = cur_cache_basefile.with_suffix(".pkl")
+        _write_pickle_file(cur_pkl_path, "wb", kernel_launch_info)
```

Comment:
Is this to pickle the dataclass of?
```python
    grid: tuple[int] = None
    blocks: tuple[int] = None
    shared_memory_bytes: int = 0
    func_name: str = ""
```
If yes, any thoughts on using jsons instead? I think json serialization and deserialization is faster thank python pickle + could be easier to debug and move around?

Reply:
Actually wrt speed of serialization and deseralization after python3 it doesn't seem to be the case anymore since it uses cPickle by default. But I think json is still worth considering for it's portability and debugability . :)

Reply:
I am not particularly tied to pickle, it was the easiest thing to use at the time. I think speed of pickle may even be faster than json, but I can change it to make it more consistent (I believe we are not using pickle anywhere else in the codebase) and for debuggability.

Reply:
Done.